src/main/java/org/jenkinsci/plugins/configfiles/buildwrapper/ManagedFile.java

```diff
@@ -106,10 +108,12 @@ public String getDisplayName() {
             return "";
         }
 
-        public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project) {
+        public ListBoxModel doFillFileIdItems(@AncestorInPath ItemGroup context, @AncestorInPath Item project, @AncestorInPath AccessControlled ac) {
             // You should have permission to configure your project in order to get the available managed files
             if (project != null) {
                 project.checkPermission(Item.CONFIGURE);
+            } else {
+                ac.checkPermission(Item.CREATE); // This to get the first parent to validate the authorization (Folder, View or Jenkins)
```

Comment:
`checkPermission` calls may end up messing with the UI because those are errors. What happens when someone opens the top-level `pipeline-syntax/` page without this permission?

Having an empty list or obvious placeholder would probably be nicer.

Reply:
This is a great idea. I don't know if I can tackle this right now but let me track this as an enhancement for the plugin. Is that ok for you?

Reply:
Sure, not a maintainer 😁

Reply:
https://issues.jenkins.io/browse/JENKINS-67202https://issues.jenkins.io/browse/JENKINS-67202